hypershift/hosted-wrapper.py

```diff
@@ -0,0 +1,592 @@
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import argparse
+import time
+import subprocess
+import sys
+import shutil
+import os
+import uuid
+import json
+import requests
+import urllib
+import logging
+import configparser
+from distutils import version
+import threading
+from git import Repo
+from libs import common
+from libs import parentParsers
+
+def _verify_cmnds(ocm_cmnd, hypershift_cmnd, my_path):
+    # If the command path was not given, download latest binary from github
+    if ocm_cmnd is None:
+        logging.info('ocm command not provided')
+        logging.info('Downloading latest binary ocm from https://github.com/openshift-online/ocm-cli/releases/')
+        tags_list = []
+        try:
+            tags = requests.get(url='https://api.github.com/repos/openshift-online/ocm-cli/git/refs/tags')
+        except (requests.ConnectionError,urllib.error.HTTPError) as err:
+            logging.error('Cannot download tags list from https://api.github.com/repos/openshift-online/ocm-cli/git/refs/tags')
+            logging.error(err)
+            return 1
+        # Get all tags, sort and use the last one
+        for tag in tags.json():
+            tags_list.append(tag['ref'].split('/')[-1].split('v')[-1])
+        last_version = sorted(tags_list,key=version.StrictVersion)[-1]
+        logging.info('Identified latest release as %s' % last_version)
+        url = 'https://github.com/openshift-online/ocm-cli/releases/download/v' + last_version + '/ocm-linux-amd64'
+        with urllib.request.urlopen(url) as response, open(my_path + '/ocm', 'wb') as out_file:
+            shutil.copyfileobj(response, out_file)
+        os.chmod(my_path + '/ocm', 0o777)
+        ocm_cmnd = my_path + "/ocm"
+    logging.info('Testing ocm command with: ocm -h')
+    ocm_cmd = [ocm_cmnd, "-h"]
+    ocm_process = subprocess.Popen(ocm_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    ocm_stdout,ocm_stderr = ocm_process.communicate()
+    if ocm_process.returncode != 0:
+        logging.error('%s unable to execute -h' % ocm_cmnd)
+        logging.error(ocm_stderr.strip().decode("utf-8"))
+        return 1
+    logging.info('ocm command validated with -h. Directory is %s' % my_path)
+    if hypershift_cmnd is None:
+        logging.info('hypershift command not provided')
+        logging.info('Cloning hypershift repo and compiling cli from https://github.com/openshift/hypershift/')
+        Repo.clone_from("https://github.com/openshift/hypershift.git", my_path + '/hypershift')
```

Comment:
Need a way to specify the version/tag of the CLI.

Reply:
ok, it could be useful, I will add a parameter to specify hypershift cli version

Reply:
add parameters to select the ocm release and the hypershift branch